package-testing/sdk-test-runner/src/protocol.ts

```diff
@@ -0,0 +1,182 @@
+import axios from 'axios';
+import { AssignmentRequest } from './dto/assignmentRequest';
+import { BanditActionRequest } from './dto/banditActionRequest';
+import { TestResponse } from './dto/testResponse';
+import { Server, Socket } from 'socket.io';
+import { createServer } from 'http';
+import { green, log } from './logging';
+import { EventEmitter } from 'stream';
+
+const assignmentPath = '/flags/v1/assignment';
+const banditPath = '/bandits/v1/action';
+const resetPath = '/sdk/reset';
+
+/**
+ * Wrapper for calls to an SDK relay
+ */
+export type SDKRelay = {
+  isReady(timeout: number): Promise<SDKInfo | SDKConnectionFailure>;
+  reset(): Promise<void>;
+  getAssignment(request: AssignmentRequest): Promise<TestResponse>;
+  getBanditAction(request: BanditActionRequest): Promise<TestResponse>;
+  close(): void;
+};
+
+export type SDKInfo = {
+  sdkName: string;
+  supportsBandits: boolean;
+};
+export type SDKConnectionFailure = {
+  errorMessage?: string;
+};
+
+/**
+ * Uses `axios` to communicate with a Server SDK Relay via http
+ */
+export class ServerSDKRelay implements SDKRelay {
+  private sdkRelayAddress: string;
+  private readonly sdkInfo: SDKInfo;
+
+  constructor(serverAddress: string, sdkName: string) {
+    this.sdkRelayAddress = serverAddress;
+
+    // All servers support bandits.
+    // In the future, we can have an endpoint for the SDK relays that specifies
+    // more sdk info like version and future capabilities.
+    this.sdkInfo = { sdkName, supportsBandits: true };
+  }
+
+  close(): void {
+    return; // nothing to do.
+  }
+
+  async reset(): Promise<void> {
+    const result = await axios.post(`${this.sdkRelayAddress}/sdk/reset`, {});
+    if (result.status != 200) {
+      throw new Error(`API Server returned unexpected status: ${result.status}`);
+    }
+  }
+
+  async getBanditAction(request: BanditActionRequest): Promise<TestResponse> {
+    const result = await axios.post(`${this.sdkRelayAddress}${banditPath}`, request);
+    return result.data as TestResponse;
+  }
+
+  isReady(): Promise<SDKInfo | SDKConnectionFailure> {
+    return Promise.resolve(this.sdkInfo);
+  }
+
+  async getAssignment(request: AssignmentRequest): Promise<TestResponse> {
+    return axios.post(`${this.sdkRelayAddress}${assignmentPath}`, request).then((result) => {
+      return result.data as TestResponse;
+    });
+  }
+}
+
+/**
+ * Uses `socket.io` to communicate with a Client SDK relay
+ */
+export class ClientSDKRelay implements SDKRelay {
+  emitter = new EventEmitter();
+
+  private isReadyPromise: Promise<SDKInfo>;
+  private _isReady = false;
```

Comment:
recognize the underscore is to deconflict with the getter. Consider also just naming it `ready`